src/renderer/components/Login.js

```diff
@@ -35,15 +43,12 @@ class Login extends React.Component {
       sendPw: '',
       sendServer: '',
       sendPort: '',
-      sendSecurity: '',
-      showAdvanced: showAdvanced
+      sendSecurity: ''
     }
   }
 
-  handleChange (event) {
-    var state = {}
-    state[event.target.id] = event.target.value
-    this.setState(state)
+  handleCredentialsChange (event) {
+    this.setState({ credentials: { [event.target.id]: event.target.value } })
```

Comment:
this makes me nervous for some reason, can we verify that these keys are within a range of accepted keys? Sorry to be nit=picky, but this seems like an important part of the process and i'd worry that somehow the event.target.id/value could be manipulated.. any thoughts?

Reply:
If we assume they can be manipulated. Would it matter?